### PR TITLE
Cb/delete comment

### PR DESCRIPTION
**Pull request by chrisbrinkmann**
### Description

Add delete comment api route and tests

### Changes

Add delete comment api route and tests

### Verification

image

image

### Trello Link

https://trello.com/c/cu3HqNl7

### Notes

Only the comment creator or owner of the parent post can delete a comment.

### CODE REVIEW

File: src/routes/comments.js [hunk 1]
```diff
@@ -40,9 +42,9 @@ router.post(
 
       res.status(201).json(comment)
     } catch (err) {
-      if (err.kind === 'ObjectId') {
-        // this message will trigger iff err is from an invalid post id
-        return res.status(404).json({ msg: 'Post not found' })
+      if (err.message.match(/^(invalid input syntax for integer)/)) {
+        // this message will trigger iff non integer is put in req param
```
Comment (Formless) [inline src/routes/comments.js]: if*

File: src/routes/comments.js [hunk 2]
```diff
@@ -60,4 +62,34 @@ router.get('/', async (req, res) => {
...
+      return res.status(401).json({ msg: 'Cannot delete other users comments' })
+    }
+
+    // delete the comment from the db
+    await comment.destroy({ force: true })
+
+    res.status(200).json({ msg: 'Comment deleted' })
+  } catch (err) {
+    if (err.message.match(/^(invalid input syntax for integer)/)) {
+      // this message will trigger iff non integer is put in req param
```
Comment (Formless) [inline src/routes/comments.js]: if*

Comment (chrisbrinkmann) [reply src/routes/comments.js]: @Formless iff = if and only if. don't you remember your discreet math?

Comment (chrisbrinkmann) [reply src/routes/comments.js]: discrete* lol

**Comment by chrisbrinkmann** [comment]
@Formless I patched the route and the tests to allow not just the commenter, but also the post owner to delete comments. Should I squash it down to 2 commits and repush or is it okay to have the patch as separate commit?

**Comment by chrisbrinkmann** [comment]
These are gonna have to stay as separate commits I think. The changes to the routes and the tests are not sequential, so I can't easily squash them together. I'm going to merge it.

**Pull request merged**